src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts

```diff
@@ -378,11 +384,19 @@ export class OsdListComponent extends ListWithDetails implements OnInit {
         osd.stats_history.in_bytes = osd.stats_history.op_in_bytes.map((i: string) => i[1]);
         osd.stats.usage = osd.stats.stat_bytes_used / osd.stats.stat_bytes;
         osd.cdIsBinary = true;
+        this.get_osd_device_count(osd.id);
+        osd.device_count = this.device_count;
         return osd;
       });
     });
   }
 
+  get_osd_device_count(id: number) {
```

Comment:
I think this is a better solution, also because you are calling an observable within this function.
```suggestion
  get_osd_device_count(osd: object) {
    this.osdService.getDevices(osd.id).subscribe((data: any[]) => {
      osd.device_count = data.length;
    });
  }
```